### PR TITLE
Android APP LINK

### PR DESCRIPTION
**Pull request by sojanpr**
Added Changes to configure and Handle Android App links
 Read me is updated to show how to configure app for Android App Link

@aaustin @derrickstaten

### CODE REVIEW

**Comment by derrickstaten** [comment]
@sojanpr have you tested whether the SDK/backend handle this link properly? While we don't serve the `.well-known` file, the API service should respond appropriately if you pass up the app link

File: Branch-SDK/src/io/branch/referral/Branch.java [hunk 1]
```diff
@@ -3255,6 +3262,12 @@ public boolean onSearchRequested() {
             return callback_.onSearchRequested();
         }
 
+        @TargetApi(Build.VERSION_CODES.M)
+        @Override
+        public boolean onSearchRequested(SearchEvent searchEvent) {
```
Comment (aaustin) [inline Branch-SDK/src/io/branch/referral/Branch.java]: what is this and the one below?

**Comment by aaustin** [comment]
this all looks good to me! glad it was easy.

File: Branch-SDK/src/io/branch/referral/Branch.java (hunk 1 above)
Comment (sojanpr) [inline Branch-SDK/src/io/branch/referral/Branch.java]: SDK is built with Android 23 , which has 2 new methods added for Branch Window Callback.

**Comment by aaustin** [comment]
:+1:

**Pull request closed without merging**